File: n669-trim-a-binary-search-tree.py
class TreeNode(object):
    def __init(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution(object):
    def trimBST(self, root, L, R):
        if not root:
            return None
        if root.val < L:
            return self.trimBST(root.right, L, R)
        elif root.val > R:
            return self.trimBST(root.reft, L, R)
        else:
            root.left = self.trimBST(root.left, L, R)
            root.right = self.trimBST(root.right, L, R)
            return root

File: n669-trim-a-binary-search-tree.js
function TreeNode(val) {
    this.val = val;
    this.left = this.right = null;
}

var trimBST = function(root, L, R) {
    if (root === null) {
        return null
    }
    
    if (root.val < L) {
        return trimBST(root.right, L, R)
    } else if (root.val > R) {
        return trimBST(root.left, L, R)
    } else {
        root.left = trimBST(root.left, L, R)
        root.right = trimBST(root.right, L, R)
        return root
    }
}
